spi_dma example not compiling?

If I try to compile the [spi dma example](https://github.com/modm-io/modm/tree/develop/examples/nucleo_f439zi/spi_dma), I get the following error:

```
$ lbuild build && scons
# [...]
In file included from modm/src/modm/platform/spi/spi_master_1_dma.hpp:97,
                 from modm/src/modm/platform.hpp:38:
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp: In instantiation of 'static modm::ResumableResult<void> modm::platform::SpiMaster1_Dma<DmaChannelRx, DmaChannelTx>::transfer(const uint8_t*, uint8_t*, std::size_t) [with DmaChannelRx = modm::platform::DmaController<2>::Channel<modm::platform::DmaBase::Channel::Stream0>; DmaChannelTx = modm::platform::DmaController<2>::Channel<modm::platform::DmaBase::Channel::Stream3>; uint8_t = unsigned char; std::size_t = unsigned int]':
main.cpp:47:4:   required from here
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp:99:16: error: 'uint8_t modm::platform::SpiMaster1::state' is private within this context
   99 |         switch(state & Bit1)
      |                ^~~~~
In file included from modm/src/modm/platform.hpp:37:
modm/src/modm/platform/spi/spi_master_1.hpp:41:24: note: declared private here
   41 |         static uint8_t state;
      |                        ^~~~~
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp:99:16: error: 'uint8_t modm::platform::SpiMaster1::state' is private within this context
   99 |         switch(state & Bit1)
      |                ^~~~~
modm/src/modm/platform/spi/spi_master_1.hpp:41:24: note: declared private here
   41 |         static uint8_t state;
      |                        ^~~~~
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp:103:17: error: 'uint8_t modm::platform::SpiMaster1::state' is private within this context
  103 |                 state |= Bit1;
      |                 ^~~~~
modm/src/modm/platform/spi/spi_master_1.hpp:41:24: note: declared private here
   41 |         static uint8_t state;
      |                        ^~~~~
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp:103:17: error: 'uint8_t modm::platform::SpiMaster1::state' is private within this context
  103 |                 state |= Bit1;
      |                 ^~~~~
modm/src/modm/platform/spi/spi_master_1.hpp:41:24: note: declared private here
   41 |         static uint8_t state;
      |                        ^~~~~
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp:147:17: error: 'uint8_t modm::platform::SpiMaster1::state' is private within this context
  147 |                 state &= ~Bit1;
      |                 ^~~~~
modm/src/modm/platform/spi/spi_master_1.hpp:41:24: note: declared private here
   41 |         static uint8_t state;
      |                        ^~~~~
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp:147:17: error: 'uint8_t modm::platform::SpiMaster1::state' is private within this context
  147 |                 state &= ~Bit1;
      |                 ^~~~~
modm/src/modm/platform/spi/spi_master_1.hpp:41:24: note: declared private here
   41 |         static uint8_t state;
      |                        ^~~~~
scons: *** [/home/user/projects/modm_projects/modm/build/nucleo_f439zi/spi_dma/scons-release/main.o] Error 1
scons: building terminated because of errors.
```
Build with recent develop branch and arm-none-eabi-gcc 12.1.0-1 on arch.
Am I missing something? CI seems to build the example...

@rleh I can reproduce this with the 2022.02 GCC 11.2 ARM toolchain but the 10.2 version works for me.

```
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp: In instantiation of 'static modm::ResumableResult<void> modm::platform::SpiMaster1_Dma<DmaChannelRx, DmaChannelTx>::transfer(const uint8_t*, uint8_t*, std::size_t) [with DmaChannelRx = modm::platform::DmaController<2>::Channel<modm::platform::DmaBase::Channel::Stream0>; DmaChannelTx = modm::platform::DmaController<2>::Channel<modm::platform::DmaBase::Channel::Stream3>; uint8_t = unsigned char; std::size_t = unsigned int]':
main.cpp:47:4:   required from here
modm/src/modm/platform/spi/spi_master_1_dma_impl.hpp:99:16: error: 'uint8_t modm::platform::SpiMaster1::state' is private within this context
   99 |         switch(state & Bit1)
      |                ^~~~~
```

--- a/src/modm/platform/spi/rp/spi_master.hpp.in
+++ b/src/modm/platform/spi/rp/spi_master.hpp.in
@@ -27,9 +27,11 @@ namespace modm::platform
  */
 class SpiMaster{{ id }} : public modm::SpiMaster
 {
+protected:
 	// Bit0: single transfer state
 	// Bit1: block transfer state
 	static inline uint8_t state{0};
+private:
 	static inline uint8_t count{0};
 	static inline void *context{nullptr};
 	static inline ConfigurationHandler configuration{nullptr};

--- a/src/modm/platform/spi/stm32/spi_master.cpp.in
+++ b/src/modm/platform/spi/stm32/spi_master.cpp.in
@@ -16,21 +16,6 @@
 
 #include "spi_master_{{id}}.hpp"
 
-// Bit0: single transfer state
-// Bit1: block transfer state
-uint8_t
-modm::platform::SpiMaster{{ id }}::state(0);
-
-uint8_t
-modm::platform::SpiMaster{{ id }}::count(0);
-
-void *
-modm::platform::SpiMaster{{ id }}::context(nullptr);
-
-modm::Spi::ConfigurationHandler
-modm::platform::SpiMaster{{ id }}::configuration(nullptr);
-// ----------------------------------------------------------------------------
-
 uint8_t
 modm::platform::SpiMaster{{ id }}::acquire(void *ctx, ConfigurationHandler handler)
 {

--- a/src/modm/platform/spi/stm32/spi_master.hpp.in
+++ b/src/modm/platform/spi/stm32/spi_master.hpp.in
@@ -5,6 +5,7 @@
  * Copyright (c) 2012, Georgi Grinshpun
  * Copyright (c) 2013, Kevin Läufer
  * Copyright (c) 2014, Sascha Schade
+ * Copyright (c) 2022, Christopher Durand
  *
  * This file is part of the modm project.
  *
@@ -38,10 +39,15 @@ namespace platform
  */
 class SpiMaster{{ id }} : public modm::SpiMaster
 {
-	static uint8_t state;
-	static uint8_t count;
-	static void *context;
-	static ConfigurationHandler configuration;
+protected:
+	// `state` must be protected to allow access from SpiMasterDma subclass
+	// Bit0: single transfer state
+	// Bit1: block transfer state
+	static inline uint8_t state{0};
+private:
+	static inline uint8_t count{0};
+	static inline void* context{nullptr};
+	static inline ConfigurationHandler configuration{nullptr};
 public:
 	using Hal = SpiHal{{ id }};
 

